pkg/cli/cli.go

```diff
@@ -121,6 +121,55 @@ func Parse(p parser.Parser, path string, cmd *cobra.Command) (parser.Result, err
 
 	result.TestResults = append(result.TestResults, testResults)
 
+	trimStdoutTo, err := cmd.Flags().GetInt32("trim-stdout-to")
+	if err != nil {
+		logger.Error("Reading flag trim-stdout-to failed with error: %v", err)
+		return result, err
+	}
+
+	if trimStdoutTo > 0 {
+		for idx := range result.TestResults {
+			for suiteIdx := range result.TestResults[idx].Suites {
+				if len(result.TestResults[idx].Suites[suiteIdx].SystemErr) > int(trimStdoutTo) {
+					result.TestResults[idx].Suites[suiteIdx].SystemErr = result.TestResults[idx].Suites[suiteIdx].SystemErr[:trimStdoutTo]
+				}
+
+				if len(result.TestResults[idx].Suites[suiteIdx].SystemOut) > int(trimStdoutTo) {
+					result.TestResults[idx].Suites[suiteIdx].SystemOut = result.TestResults[idx].Suites[suiteIdx].SystemOut[:trimStdoutTo]
+				}
+
+				for caseIdx := range result.TestResults[idx].Suites[suiteIdx].Tests {
+					if len(result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemErr) > int(trimStdoutTo) {
+						result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemErr = result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemErr[:trimStdoutTo]
+					}
+
+					if len(result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemOut) > int(trimStdoutTo) {
+						result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemOut = result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemOut[:trimStdoutTo]
+					}
+				}
+			}
+		}
+	}
+
+	omitStdoutForPassed, err := cmd.Flags().GetBool("omit-stdout-for-passed")
+	if err != nil {
+		logger.Error("Reading flag omit-stdout-for-passed failed with error: %v", err)
+		return result, err
+	}
+
+	if omitStdoutForPassed {
```

Comment:
Maybe checking this flag first is better. If this is specified, we don't even need to look into the other one, avoiding trimming long string sequences that will just be removed in the end.